### PR TITLE
Change using posix-threads to utils-threads

### PR DESCRIPTION
**Pull request by OHerasym**
### PR changes: ###
Change posix threads to utils-threads in:
 - conditional_variable_test.cc
 - rwlock_test.cc
 - singleton_test.cc
 - message_queue_test.cc

> Tests builds on LINUX & WINDOWS & QT platform

Related: [APPLINK-24859](https://adc.luxoft.com/jira/browse/APPLINK-24859)

Previous [pull request](https://github.com/anosach-luxoft/sdl_core/pull/34)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @anosach-luxoft

**Comment by anosach-luxoft** [comment]
@OHerasym add all reviewers

**Comment by OHerasym** [comment]
Please review: @Kozoriz , @LuxoftAKutsan , @dtrunov

File: src/components/utils/test/conditional_variable_test.cc [hunk 1]
```diff
@@ -38,32 +38,45 @@
...
 namespace components {
 namespace utils_test {
 
 class ConditionalVariableTest : public ::testing::Test {
  public:
-  ConditionalVariableTest() : test_value_("initialized"), counter_(0) {}
-  void check_counter();
-  void task_one();
+  ConditionalVariableTest() {}
+  static void check_counter();
```
Comment (LuxoftAKutsan) [inline src/components/utils/test/conditional_variable_test.cc]: @OHerasym  unclear method purpose. Please Add doxygen comment and change to CamelCase

File: src/components/utils/test/message_queue_test.cc [hunk 2]
```diff
@@ -42,92 +43,108 @@ using ::utils::MessageQueue;
...
+  void exitThreadMain() {}
+};
 
 TEST_F(MessageQueueTest, DefaultCtorTest_ExpectEmptyQueueCreated) {
   bool test_value = true;
   // Check if the queue is empty
   ASSERT_EQ(test_value, test_queue.empty());
 }
 
+class A : public threads::ThreadDelegate {
```
Comment (LuxoftAKutsan) [inline src/components/utils/test/message_queue_test.cc]: @OHerasym  class A? Please name it properly

**Comment by dtrunov** [comment]
Reviewed

**Comment by LuxoftAKutsan** [comment]
Reviewed

File: src/components/utils/test/conditional_variable_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/utils/test/conditional_variable_test.cc]: @LuxoftAKutsan , af35771 , Done

File: src/components/utils/test/message_queue_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/utils/test/message_queue_test.cc]: @LuxoftAKutsan , af35771 , Forgot to Remove

**Comment by Kozoriz** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
Reviewed

**Pull request merged**